gdal/gcore/overview.cpp

```diff
@@ -526,16 +528,24 @@ GDALResampleChunk32R_AverageT( double dfXRatioDstToSrc,
             {
                 // Optimized case : no nodata, overview by a factor of 2 and
                 // regular x and y src spacing.
+                Tsum nTotal;
                 const T* pSrcScanlineShifted =
                     pChunk + panSrcXOffShifted[0] +
                     static_cast<GPtrDiff_t>(nSrcYOff - nChunkYOff) * nChunkXSize;
                 for( int iDstPixel = 0; iDstPixel < nDstXWidth; ++iDstPixel )
                 {
-                    const Tsum nTotal =
-                        pSrcScanlineShifted[0]
-                        + pSrcScanlineShifted[1]
-                        + pSrcScanlineShifted[nChunkXSize]
-                        + pSrcScanlineShifted[1+nChunkXSize];
+                    if(bQuadraticAverage)
+                        nTotal =
+                            pSrcScanlineShifted[0]*pSrcScanlineShifted[0]
```

Comment:
You should try to follow existing style of whitespacing: `pSrcScanlineShifted[0] * pSrcScanlineShifted[0]`

Reply:
ok